### PR TITLE
karpenter: upgrade to version 0.16.3 and support kubeReserved configuration

### PR DESCRIPTION
**Pull request by noony**
Hi,

This pr upgrade karpenter to the version 0.16.3 and add support of kubeReserved configuration regarding the instance group config.

It will help to have a better karpenter consolidation, a better karpenter instance choice by not spawning a too small node regarding kubelet and kubernetes reservations.

Thanks.

### CODE REVIEW

File: upup/models/cloudup/resources/addons/karpenter.sh/k8s-1.19.yaml.template [hunk 1]
```diff
@@ -1154,6 +1204,12 @@ spec:
       {{ $key }}: "{{ $val }}"
       {{ end }}
     {{ end }}
+    {{ if .KubeReserved }}
+    systemReserved:
```
Comment (olemarkus) [inline upup/models/cloudup/resources/addons/karpenter.sh/k8s-1.19.yaml.template:1208]:
```suggestion
    kubeReserved:
```

**Review by olemarkus — CHANGES_REQUESTED** [review]
/ok-to-test

File: pkg/model/components/addonmanifests/karpenter/iam.go [hunk 2]
```diff
@@ -56,6 +56,7 @@ func addKarpenterPermissions(p *iam.Policy) {
 		"ec2:DescribeAvailabilityZones",
 		"ec2:DescribeInstanceTypeOfferings",
 		"ec2:DescribeInstanceTypes",
+		"ec2:DescribeImages",
```
[the remark this answers is not in the document]
Comment (noony) [reply pkg/model/components/addonmanifests/karpenter/iam.go]: You are right following the documentation, I removed it

File: upup/models/cloudup/resources/addons/karpenter.sh/k8s-1.19.yaml.template (hunk 1 above)
Comment (noony) [reply upup/models/cloudup/resources/addons/karpenter.sh/k8s-1.19.yaml.template]: fixed

File: tests/integration/update_cluster/karpenter/in-v1alpha2.yaml [hunk 3]
```diff
@@ -116,6 +116,9 @@ spec:
     systemReserved:
       cpu: 500m
       memory: 1G
+    kubeReserved:
```
[the remark this answers is not in the document]
Comment (noony) [reply tests/integration/update_cluster/karpenter/in-v1alpha2.yaml:119]: it is now it's was due to the typo

[the remark this answers is not in the document]
Comment (noony) [reply tests/integration/update_cluster/karpenter/in-v1alpha2.yaml:119]: cf https://github.com/kubernetes/kops/pull/14408/files#diff-0e9477ea0a1226905c4ef323e8de0b933dfb6ecd989b26d38c67981501500a98R54

[the remark this answers is not in the document]
Comment (noony) [reply tests/integration/update_cluster/karpenter/in-v1alpha2.yaml:119]: cf : https://github.com/kubernetes/kops/pull/14408/files#diff-eb2250449a3edeb078a19e4df68846aef9059514571d40934bcd3d526015bc30R1460

**Review by olemarkus — APPROVED** [review]
Thanks a lot for this!

**Pull request merged**